dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/ImplementableRuleService.java

```diff
@@ -77,7 +78,14 @@ public ImplementableRuleService( ProgramRuleService programRuleService )
 
         List<ProgramRule> programRulesByActionTypes = getProgramRulesByActionTypes( program, programStageUid );
 
-        getProgramRulesCache().put( program.getUid(), !programRulesByActionTypes.isEmpty() );
```

Comment:
I think this cache could have been called `programHasRulesCache` to make it more descriptive.

Reply:
resolved